lightning/src/routing/gossip.rs

```diff
@@ -628,15 +628,77 @@ impl fmt::Display for ChannelUpdateInfo {
 	}
 }
 
-impl_writeable_tlv_based!(ChannelUpdateInfo, {
-	(0, last_update, required),
-	(2, enabled, required),
-	(4, cltv_expiry_delta, required),
-	(6, htlc_minimum_msat, required),
-	(8, htlc_maximum_msat, required),
-	(10, fees, required),
-	(12, last_update_message, required),
-});
+impl Writeable for ChannelUpdateInfo {
+	fn write<W: ::util::ser::Writer>(&self, writer: &mut W) -> Result<(), io::Error> {
+		write_tlv_fields!(writer, {
+			(0, self.last_update, required),
+			(2, self.enabled, required),
+			(4, self.cltv_expiry_delta, required),
+			(6, self.htlc_minimum_msat, required),
+			(8, self.htlc_maximum_msat, required),
+			(10, self.fees, required),
+			(12, self.last_update_message, required),
+		});
+		Ok(())
+	}
+
+	#[inline]
+	fn serialized_length(&self) -> usize {
+		use util::ser::BigSize;
+		let len = {
+			#[allow(unused_mut)]
+			let mut len = ::util::ser::LengthCalculatingWriter(0);
+			get_varint_length_prefixed_tlv_length!(len, 0, self.last_update, required);
+			get_varint_length_prefixed_tlv_length!(len, 2, self.enabled, required);
+			get_varint_length_prefixed_tlv_length!(len, 4, self.cltv_expiry_delta, required);
+			get_varint_length_prefixed_tlv_length!(len, 6, self.htlc_minimum_msat, required);
+			get_varint_length_prefixed_tlv_length!(len, 8, self.htlc_maximum_msat, required);
+			get_varint_length_prefixed_tlv_length!(len, 10, self.fees, required);
+			get_varint_length_prefixed_tlv_length!(len, 12, self.last_update_message, required);
+			len.0
+		};
+		let mut len_calc = ::util::ser::LengthCalculatingWriter(0);
+		BigSize(len as u64).write(&mut len_calc).expect("No in-memory data may fail to serialize");
+		len + len_calc.0
+	}
+
+}
+
+impl MaybeReadable for ChannelUpdateInfo {
+	fn read<R: io::Read>(reader: &mut R) -> Result<Option<Self>, DecodeError> {
+		init_tlv_field_var!(last_update, required);
```

Comment:
Agreed. Currently they are all `OptionDeserWrapper`, but will do this when I end up doing the reads manually.

Reply:
Yes, because there are quite a few where I'd have to use the `OptionDeserWrapper` anyways, e.g., due to some trait bound requirements or for example since there isn't a good default value for `NodeId`. So I figured utilizing the macro notation at least makes it uniform and rather obvious what fields have the default behavior and which ones are changed in this implementation. Let me know if I really really should avoid using the macros here.